models/opnilog/opnilog_parser.py

```diff
@@ -366,9 +368,10 @@ def run_epoch(self, dataloader, model, loss_compute):
 
             if i % self.step_size == 1:
                 elapsed = time.time() - start
+                training_progress = (i / len(dataloader) + epoch) / self.nr_epochs
+                total_time_taken = time.time() - training_start_time
                 logging.info(
-                    "Epoch Step: %d / %d Loss: %f Tokens per Sec: %f"
-                    % (i, len(dataloader), loss / batch.ntokens, tokens / elapsed)
+                    f"| Epoch: {epoch} | Total Progress: {(training_progress * 100):.2f}% | Training Time Taken: {total_time_taken:.2f}s | ETC: {(total_time_taken // training_progress)}s | Epoch Step: {i}/{len(dataloader)} | Loss: {(loss / batch.ntokens):.4f} | Tokens per Sec: {(tokens / elapsed):.2f} |"
```

Comment:
While its okay to log the information, this should also be sent over Nats and the training controller should have a co-routine that listens to the Nats subject and updates its information accordingly.